server/apps/main/views.py

```diff
@@ -502,3 +509,22 @@ def moderate_experience(request, uuid):
             )
     else:
         return redirect("index")
+
+
+def single_story(request, uuid):
+    """
+    Returns a single page with one story on it
+    """
+    # Must have both the specified UUID and be approved otherwise will redirect
+    # Should only be one result if not redirect
+    try:
+        experiences = PublicExperience.objects.filter(
+            experience_id=uuid, moderation_status="approved"
+        )
+        title = experiences.values_list("title_text", flat=True)[0]
+        exp_context = {"experiences": experiences}
+        title_context = {"title": title}
+        context = {**exp_context, **title_context}
+        return render(request, "main/single_story.html", context=context)
+    except:
+        return redirect("main:overview")
```

Comment:
```suggestion
    try:
        experience = PublicExperience.objects.get(
            experience_id=uuid, moderation_status="approved"
        )
        title = experience.title_text
        context = {
            "experience": experience,
            "title": title
            }
        return render(request, "main/single_story.html", context=context)
    except ObjectDoesNotExist:
        return redirect("main:overview")

```

This would be my suggestion for how to handle it, it tries to `get` the single experience based on the ID & moderation status and render the HTML. If it excepts because of the object not existing redirect. 

This change also handles a single experience to the template for rendering, so could remove the for-loop in the template (see above).